tests/api2/activedirectory.py

```diff
@@ -0,0 +1,563 @@
+                                       ]) is False, reason=BSDReason)
+
+osx_host_cfg = pytest.mark.skipif(all(["OSX_HOST" in locals(),
+                                       "OSX_USERNAME" in locals(),
+                                       "OSX_PASSWORD" in locals()
+                                       ]) is False, reason=OSXReason)
+
+
+def test_01_get_activedirectory_data():
+    global results
+    results = GET('/activedirectory/')
+    assert results.status_code == 200, results.text
+
+
+@pytest.mark.parametrize('data', list(ad_data_type.keys()))
+def test_02_verify_activedirectory_data_type_of_the_object_value_of_(data):
+    assert isinstance(results.json()[data], ad_data_type[data]), results.text
+
+
+def test_03_get_activedirectory_state():
+    results = GET('/activedirectory/get_state/')
+    assert results.status_code == 200, results.text
+    assert results.json() == 'DISABLED', results.text
+
+
+def test_04_get_activedirectory_started_before_starting_activedirectory():
+    results = GET('/activedirectory/started/')
+    assert results.status_code == 422, results.text
+
+
+def test_05_creating_ad_dataset_for_smb():
+    results = POST("/pool/dataset/", {"name": dataset})
+    assert results.status_code == 200, results.text
+
+
+def test_06_Changing_permissions_on_dataset():
+    results = POST(f'/pool/dataset/id/{dataset_url}/permission/', {
+        'acl': [],
+        'mode': '777',
+        'user': 'root',
+        'group': 'wheel'
+    })
+    assert results.status_code == 200, results.text
+
+
+@ad_test_cfg
+def test_07_enabling_activedirectory():
+    global payload, results
+    payload = {
+        "bindpw": ADPASSWORD,
+        "bindname": ADUSERNAME,
+        "domainname": BRIDGEDOMAIN,
+        "netbiosname": BRIDGEHOST,
+        "idmap_backend": "RID",
+        "enable": True
+    }
+    results = PUT("/activedirectory/", payload)
+    assert results.status_code == 200, results.text
+
+
+@ad_test_cfg
+def test_08_get_activedirectory_state():
+    global results
+    results = GET('/activedirectory/get_state/')
+    assert results.status_code == 200, results.text
+    assert results.json() == 'HEALTHY', results.text
+
+
+@ad_test_cfg
+def test_09_get_activedirectory_started():
+    results = GET('/activedirectory/started/')
+    assert results.status_code == 200, results.text
+    assert results.json() is True, results.text
+
+
+@ad_test_cfg
+def test_10_get_activedirectory_data():
+    global results
+    results = GET('/activedirectory/')
+    assert results.status_code == 200, results.text
+
+
+@ad_test_cfg
+@pytest.mark.parametrize('data', ad_object_list)
+def test_11_verify_activedirectory_data_of_(data):
+    if data == 'domainname':
+        assert results.json()[data].lower() == payload[data], results.text
+    else:
+        assert results.json()[data] == payload[data], results.text
+
+
+def test_12_setting_up_smb():
+    global payload, results
+    payload = {
```

Comment:
If you're using mount_smbfs on FreeBSD, then you may need to set `enable_smb1=True`